src/etc/inc/interfaces.inc

```diff
@@ -2946,9 +2946,131 @@ function interface_dhcpv6_prepare($interface = 'wan', $wancfg)
     $wanif = get_real_interface($interface, 'inet6');
+    
+    $dhcp6cscript = "#!/bin/sh\n";
+    $dhcp6cscript .= "if [ -n '" . (!empty($wancfg['adv_dhcp6_debug']) ? 'debug' : '') . "' ]; then\n";
+    $dhcp6cscript .= "\t/usr/bin/logger -t dhcp6c \"dhcp6c \$REASON on {$wanif}\"\n";
+    $dhcp6cscript .= "fi\n";
+    $dhcp6cscript .= "case \$REASON in\n";
+    $dhcp6cscript .= "REQUEST|" . (!empty($wancfg['dhcp6norelease']) ? 'EXIT' : 'RELEASE') . ")\n";
+    $dhcp6cscript .= "\t/usr/bin/logger -t dhcp6c \"dhcp6c \$REASON on {$wanif} - running newipv6\"\n";
+    $dhcp6cscript .= "\t/usr/local/opnsense/service/configd_ctl.py interface newipv6 {$wanif}\n";
+    $dhcp6cscript .= "\t;;\n";
+    $dhcp6cscript .= "*)\n";
+    $dhcp6cscript .= "\t;;\n";
+    $dhcp6cscript .= "esac\n";
+
+    if (!@file_put_contents("/var/etc/dhcp6c_{$wanif}_script.sh", $dhcp6cscript)) {
+        log_error("Error: cannot open dhcp6c_{$wanif}_script.sh in interface_dhcpv6_configure() for writing.");
+        return;
+    }
 
-    $dhcp6cconf = "interface {$wanif} {\n";
+    chmod("/var/etc/dhcp6c_{$wanif}_script.sh", 0755);
+    
+    if(empty($wancfg['adv_dhcp6_debug']) && empty($wancfg['dhcp6norelease'])) {
+        $dhcp6ccommand = exec_safe(
+            "/usr/local/sbin/dhcp6c -c %s -p %s",
+            array(            
+                "/var/etc/dhcp6c.conf",
+                "/var/run/dhcp6c.pid",
+            )
+        );
+    } else {    
+        $dhcp6ccommand = exec_safe(
+            "/usr/local/sbin/dhcp6c %s -c %s -p %s",
+            array(
+                '-' . (empty($wancfg['adv_dhcp6_debug']) ? '' : $wancfg['adv_dhcp6_debug']) . (!empty($wancfg['dhcp6norelease']) ? 'n' : ''),
+                "/var/etc/dhcp6c.conf",
+                "/var/run/dhcp6c.pid",                
+            )
+        );
+    }
+
+    // Add the interfaces on the end, as there may be more than one interface - multiple WAN dhcp6c.
+    
+    foreach (legacy_config_get_interfaces(['enable' => true, 'virtual' => false]) as $if => $wancfg) {
+        // check the interface is up
+        $realif = get_real_interface($if,'inet6');
+        $intf_stats = legacy_interface_stats($realif);
+        if($intf_stats['link state'] != 2) {
+            continue;
+        }
+        if (!empty($wancfg['ipaddrv6']) && $wancfg['ipaddrv6'] == 'dhcp6' && $wancfg['enable'] == true) {
+            
+            $dhcp6ccommand = exec_safe("{$dhcp6ccommand} %s", get_real_interface($if, 'inet6'));
+        }
+    }
+
+    log_error("dhcp6c command = {$dhcp6ccommand}"); // debug purposes only
+
+    $rtsoldscript = "#!/bin/sh\n";
+    $rtsoldscript .= "# this file was auto-generated, do not edit\n";
+    $rtsoldscript .= "if [ -n \"\${2}\" ]; then\n";
+    $rtsoldscript .= "\techo \${2} > /tmp/{$wanif}_routerv6\n";
+    $rtsoldscript .= "\techo \${2} > /tmp/{$wanif}_defaultgwv6\n";
+    $rtsoldscript .= "fi\n";
+    $rtsoldscript .= "val=`pgrep dhcp6c`\n";
```

Comment:
moving blocks, shell code quality decline. If we need to move code fine, if we end up replacing working code, please no.

Reply:
OK, so if we don't replace code then there's no multwan dhcp6; there can be only one client and we have to make certain that that is the case. However we can have more than rtsold script, thus we have to replace the code.

Reply:
It's what I'm saying. If we don't put work into cleaning this up and/or refactoring step by step there's no multiwan dhcp6. We have tens of thousands of angry IPv6 users already who require us to not mess up single dhcp6 in the process.

Reply:
No I totally agree, baby steps. I was going to suggest we first refactor the dhcp6c stuff anyway. Shall I create a PR with just the dhcp6c stuff and we can pull that to pieces and make sure it fits the needs now and in the future, possible breaking it out into plugins.inc.d?

Reply:
Yes, please. good idea. I don't mind breaking master with the dhcp6c pid file change. Second WAN doesn't work anyway and more pressure to sync up with 20.7. What do you think?

Reply:
OK, well I've just taken the interfaces.inc from 20.7 and applied the changes needed to enable multwan there. Just completing testing and it looks good. 

BTW, I'm testing it in the following way. I have multiple VLANs coming off the primary router, each of those VLANs has a different V6 range. I then have my test router with two WANs plugged into separate VLANs and two LANs outputting to further ports on my switch. I can therefore test that the WANs are getting addresses correctly and that the LAN's are also being given the correct addresses by dhcp6c.

I realise that looking at a whole load of code changes is not nice if you don't know the thought behind it and what's related to what. I'll PR the new file now and we'll just work our way through it before moving onto the other stuff. There is t he extra addition of the debug level feature added to interfaces.php, I'll add that as well; I believe that I have worked my way around the things you didn't like when I did the original PR for that.

Reply:
PR #4082 issued. So let's work our way through the changes step by step so we all understand what each addition/change does and why.

Reply:
Remember I am on furlough at present, and apart from some webinars etc that I'm doing I have time to play with this, at least until the lockdown is over.